gsoc-2025/work-product/Prasoon-Kumar-Expanding-Software-Support-Ecosystem/Prasoon-Kumar-Expanding-Software-Support-Ecosystem.md

```diff
@@ -0,0 +1,128 @@
+# Unikraft GSoC'25: Expanding the Unikraft Software Support Ecosystem
+
+<img src="https://unikraft.org/images/gsoc25.jpeg"/>
+
+## Summary
+
+Unikraft makes it easy to run existing applications in two ways:
+1. **Musl libc support**:
+Applications can be compiled against Musl libc and linked directly with Unikraft to produce small, efficient unikernels.
+
+2. [**Binary-compatibility mode**](https://unikraft.org/docs/concepts/compatibility):
+Unikraft can run unmodified Linux ELF binaries by using an ELF loader app and a system-call shim layer.
+When a binary-compatible unikernel starts, the `app-elfloader` parses and maps the ELF segments into memory, then transfers control to the application’s entry point.
+Each time the application makes a Linux system call, the shim intercepts the call and routes it to the matching Unikraft handler.
+If a call is not implemented, it returns `ENOSYS`.
+Many applications can still handle or bypass this because Unikraft follows the Linux x86_64 ABI closely.
+This allows existing binaries to run on top of Unikraft without recompilation.
```

Comment:
```suggestion
   Unikraft can run unmodified Linux ELF binaries by using an ELF loader app and a system-call shim layer.
   When a binary-compatible unikernel starts, the `app-elfloader` parses and maps the ELF segments into memory, then transfers control to the application’s entry point.
   Each time the application makes a Linux system call, the shim intercepts the call and routes it to the matching Unikraft handler.
   If a call is not implemented, it returns `ENOSYS`.
   Many applications can still handle or bypass this because Unikraft follows the Linux x86_64 ABI closely.
   This allows existing binaries to run on top of Unikraft without recompilation.
```

Indent to fit the list item.